neo4j/v1/session.py

```diff
@@ -510,14 +512,28 @@ def __ne__(self, other):
         return not self.__eq__(other)
 
 
-def basic_auth(user, password):
+def basic_auth(user, password, realm=None):
     """ Generate a basic auth token for a given user and password.
 
     :param user: user name
     :param password: current password
+    :param realm: specifies the authentication provider
+    :return: auth token for use with :meth:`GraphDatabase.driver`
+    """
+    return AuthToken("basic", user, password, realm)
+
+
+def custom_auth(principal, credentials, realm=None, scheme=None, parameters=None):
```

Comment:
Do we want optional params for realm and scheme? Not consistent with java driver.